interview-questions.md

```diff
@@ -4,56 +4,90 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is object destructuring?
 
-  Your answer:
+  Your answer: it is a way for us to call keys in state without having to type out the full this.state. syntax everytime we nee need something.
 
-  Researched answer:
+  Researched answer: The object destructuring is a useful JS feature used to extract props from objects and bind them to variables. with object destructuring we can extract multiple properties in one statement, can access properties from nested objects, and can set a default value if the property doesn't exist.
 
 
 
 2. What is a DOM event?
 
   Your answer:
+  A DOM event is something that happens to the document object model, something like an on change or an on click.
 
   Researched answer:
-
+DOM events are actions that occurs as a result of the user action or as result of state change of the elements of a DOM tree.
 
 
 3. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: 
+  A class is react's way of storing groups of information together under one thing, it's based on objects. Objects are simply just groups of information without the functionality of react.
 
   Researched answer:
+  A class is a template for objects, an object is an instance of a class.
 
 
 
 4. What is the difference between a HTML div and a span?
 
   Your answer:
+  a div is a block level element, it takes up its own line. A span is an inline element so it doesn't create any line breaks.
 
   Researched answer:
-
+to expand on my answer, it says that divs should be used to wrap sections of a document while spans are used to wrap around small portions of text, images, etc.
 
 
 5. What is React.Fragment (or <>) and why would you want to use it?
 
-  Your answer:
+  Your answer: React.Fragment is where we put everything that we want to render on screen. All component calls and such are stored in the react fragment.
```

Comment:
Good answer. I know since you researched this a bit more, if/when you're ever asked this again,  I think you could include the part about it not requiring extra nodes in the DOM.